src/fs/filesystem.ts

```diff
@@ -145,24 +132,14 @@ export class FileSystem {
     })
   }
 
-  async pinList(): Promise<CID[]> {
-    const privateResult = await this.privateTree.putWithPins()
-    const publicResult = await this.publicTree.putWithPins()
-    const rootCID = await this.sync()
-    return [
-      ...privateResult.pins,
-      ...publicResult.pins,
-      rootCID
-    ]
+  async updatePinTree(pins: PinMap): Promise<void> {
+    const pinLinks = await pinMapToLinks('private', pins) // @@TODO: add salt
+    console.log('pinLinks: ', pinLinks)
+    this.pinTree = BareTree.fromLinks(pinLinks)
+    await this.root.addChild('pins', this.pinTree)
   }
 
   async sync(): Promise<CID> {
-    this.root = await asyncWaterfall(this.root, [
```

Comment:
`sync` was getting more expensive with each branch we added so I moved the updates to `root` to the locations in the code where a given tree was actually being updated